### PR TITLE
push for sequences

### PR DESCRIPTION
**Pull request by hwchen**
allow push on sequences, to be able to build incrementally.

### CODE REVIEW

**Comment by mkysel** [comment]
I don't see an issue with this, but I will leave the final word to @vgel or @benwr

File: src/rust_api.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@ pub trait BaseSequence: std::marker::Sized {
 
     fn as_slice(&self) -> &[Self::Item];
     fn as_mut_slice(&mut self) -> &mut [Self::Item];
+    fn push(&mut self, _item: Self::Item) {}
```
Comment (vgel) [inline src/rust_api.rs:18]: I don't like this having a default impl that does nothing. It seems error-prone.

I'd prefer that `push` on `ProteinSequence` and `DnaSequence` are just `pub fn`s in impl blocks for those types, like how `DnaSequence` already has `translate`. Sort of like how `Vec` and `BinaryHeap` both have `push`, but it's not part of a trait, just an associated method with similar semantics & name. We might want to implement `BaseSequence` for immutable view types at some point, and having a mutable function in the trait will be a blocker for that.

**Review by vgel — CHANGES_REQUESTED** [review]

File: src/rust_api.rs (hunk 1 above)
Comment (hwchen) [reply src/rust_api.rs:18]: Ah, this is a typo, I didn't intend for a default impl.

I'm fine w/ not having `push` as a part of the trait

Comment (hwchen) [reply src/rust_api.rs]: ping @vgel

**Review by vgel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**